cli/src/main/java/org/jboss/as/cli/impl/ReadlineConsole.java

```diff
@@ -471,19 +462,15 @@ private CLITerminalConnection newConnection() throws IOException {
                 .system(!settings.isOutputRedefined())
                 .build();
         CLITerminalConnection c = new CLITerminalConnection(terminal);
-        if (LOG.isLoggable(Level.FINER)) {
-            LOG.log(Level.FINER, "New Terminal {0}", terminal.getClass());
-        }
+        CliLogger.ROOT_LOGGER.newTerminal(terminal.getClass());
         return c;
     }
 
     @Override
     public void addCompleter(CommandLineCompleter completer) {
         completions.add((Completion) (CompleteOperation co) -> {
             List<String> candidates = new ArrayList<>();
-            if (LOG.isLoggable(Level.FINER)) {
-                LOG.log(Level.FINER, "Completing {0}", co.getBuffer());
-            }
+            CliLogger.ROOT_LOGGER.completingBuffer(co.getBuffer());
```

Comment:
This should probably check if trace is enabled. I'm guessing `co.getBuffer()` may be expensive.